codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/CodegenOrchestrator.java

```diff
@@ -264,14 +232,76 @@ private void generateYardOpts() {
         Optional<TitleTrait> title = context.service().getTrait(TitleTrait.class);
         if (title.isPresent()) {
             FileManifest fileManifest = context.fileManifest();
-            CodeWriter writer = new CodeWriter();
+            RubyCodeWriter writer = new RubyCodeWriter();
             writer.write("--title \"$L\"", title.get().getValue());
             writer.write("--hide-api private");
             String fileName = context.settings().getGemName() + "/.yardopts";
             fileManifest.writeFile(fileName, writer.toString());
         }
     }
 
+    private void generateTypes() {
+        TypesGenerator typesGenerator = new TypesGenerator(context);
+        typesGenerator.render();
+        typesGenerator.renderRbs();
+        LOGGER.info("generated types");
+    }
+
+    private void generateParams() {
+        ParamsGenerator paramsGenerator = new ParamsGenerator(context);
+        paramsGenerator.render();
+        LOGGER.info("generated params");
+    }
+
+    private void generateValidators() {
+        ValidatorsGenerator validatorsGenerator =
+                new ValidatorsGenerator(context);
+        validatorsGenerator.render();
+        LOGGER.info("generated validators");
+    }
+
+    private void generateConfigAndClient() {
+        // Register all middleware
+        MiddlewareBuilder middlewareBuilder = new MiddlewareBuilder();
+        middlewareBuilder.addDefaultMiddleware(context);
+
+        context.integrations().forEach((integration) -> {
+            integration.modifyClientMiddleware(middlewareBuilder, context);
+        });
+
+        context.protocolGenerator().ifPresent((g) -> g.modifyClientMiddleware(middlewareBuilder, context));
+
+        // get all config
+        Set<ClientConfig> unorderedConfig = new HashSet();
+        unorderedConfig
+                .addAll(context.applicationTransport().getClientConfig());
+        unorderedConfig.addAll(middlewareBuilder.getClientConfig(context));
+        unorderedConfig.addAll(context.integrations()
+                .stream()
+                .map((i) -> i.getAdditionalClientConfig(context))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList())
+        );
+        context.protocolGenerator().ifPresent((g) -> unorderedConfig.addAll(g.getAdditionalClientConfig(context)));
+
+        List<ClientConfig> clientConfigList = unorderedConfig.stream()
+                .sorted(Comparator.comparing(ClientConfig::getName))
+                .collect(Collectors.toList());
+
+        LOGGER.fine("Client config: "
+                + clientConfigList.stream().map((m) -> m.toString()).collect(Collectors.joining(",")));
+
+        ConfigGenerator configGenerator = new ConfigGenerator(context);
+        configGenerator.render(clientConfigList);
+        configGenerator.renderRbs();
```

Comment:
does this not need the clientConfigList?
I'd probably rather pass clientConfigList to the constructor and then have render and renderRbs not take arguments

Reply:
RBS render will not need it AFAIK

Reply:
Don't we need to know fields and types the struct has to generate RBS for it?